pyanaconda/ui/gui/spokes/root_password.py

```diff
@@ -316,36 +316,32 @@ def _check_spoke_exit_conditions(self, unwaivable_check_failed):
     def on_password_changed(self, editable, data=None):
         """Tell checker that the content of the password field changed."""
         self.checker.password.content = self.password
-        # unlock the password if user starts typing
-        self._lock.set_active(False)
 
     def on_password_confirmation_changed(self, editable, data=None):
         """Tell checker that the content of the password confirmation field changed."""
         self.checker.password_confirmation.content = self.password_confirmation
-        # unlock the password if user starts typing
-        self._lock.set_active(False)
 
     def on_password_icon_clicked(self, entry, icon_pos, event):
         """Called by Gtk callback when the icon of a password entry is clicked."""
         set_password_visibility(entry, not entry.get_visibility())
 
     def on_back_clicked(self, button):
-        # the GUI spoke input check handler handles the spoke exit logic for us
-        if self.try_to_go_back() or self._lock.get_active():
+        # disable root if no password is entered
+        if not self.password and not self.password_confirmation and \
+                not self._users_module.IsRootPasswordSet:
+            control = self._disable_radio
+            control.set_active(True)
+            self.on_disable_enable_clicked(control)
+
+        # the GUI spoke input check handler handles the rest of the spoke exit logic for us
+        if self.try_to_go_back() or not self.root_enabled:
             NormalSpoke.on_back_clicked(self, button)
         else:
             log.info("Return to hub prevented by password checking rules.")
 
-    def on_lock_clicked(self, lock):
-        if self._refresh_running:
-            # this is not a "real" click, just refresh() setting the lock check
-            # box state based on data from the DBus module
-            if not self._manually_locked:
-                # if the checkbox has not yet been manipulated by the user
-                # we can ignore this run and not adjust the fields
-                return True
-        self.password_entry.set_sensitive(not lock.get_active())
-        self.password_confirmation_entry.set_sensitive(not lock.get_active())
-        if not lock.get_active():
+    def on_disable_enable_clicked(self, control):
```

Comment:
This method name seems pretty confusing to me. Right now I'm confused if it should toggle radio buttons or disable some and enable some. Could you please change the name?

Reply:
Sorry, I don't agree. It's an event handler - the name conforms to the `on_controlname_eventtype` pattern most of our code uses. You do not call it, Gtk does. And the next line is a docstring, which also says what it is: _Click handler for enable and disable radio buttons._

Reply:
In that case is `disable_enable` a good control name?

Reply:
The handler is used by two controls, one enables, one disables.

Maybe not mentioning the control names would make the name work? Eg. `on_root_enabled_changed`

Reply:
That seems much better to understand to me.

Reply:
Great, I will do that, then.

Reply:
Done.